modules/radsBidAdapter.js

```diff
@@ -97,9 +97,46 @@ export const spec = {
       bidResponses.push(bidResponse);
     }
     return bidResponses;
+  },
+  getUserSyncs: function(syncOptions, serverResponses, gdprConsent, uspConsent) {
+    if (!serverResponses || serverResponses.length === 0) {
+      return [];
+    }
+
+    const syncs = []
+
+    let gdprParams = '';
+    if (gdprConsent) {
+      if ('gdprApplies' in gdprConsent && typeof gdprConsent.gdprApplies === 'boolean') {
+        gdprParams = `gdpr=${Number(gdprConsent.gdprApplies)}&gdpr_consent=${gdprConsent.consentString}`;
+      } else {
+        gdprParams = `gdpr_consent=${gdprConsent.consentString}`;
+      }
+    }
+
+    if (syncOptions.iframeEnabled) {
```

Comment:
This conditional should also be checking serverResponses.length. It may also help to check for the existence of 'userSync.[iframeUrl/imageUrl]' in both conditionals.

Reply:
Such a check is located above in the line 102 or what do you mean about `checking serverResponses.length`?

Reply:
If a publisher only has iframeEnabled syncs this is the only conditional that will fire off your usersyncs, you are not checking your serverResponses for length before accessing the array at zero and attempting to set the iframe url.

Reply:
``` getUserSyncs: function(syncOptions, serverResponses, gdprConsent, uspConsent) {
    if (!serverResponses || serverResponses.length === 0) {
      return [];
    }
...
if (syncOptions.iframeEnabled) {
      serverResponses[0].body.userSync.iframeUrl.forEach((url) => syncs.push({
...
```
I check the number of server responses at the beginning of the function, if there is 0 - then the function will return earlier. Is not it?

Reply:
Ok that is fine, you are checking for it on line 123 so I was curious why it wasn't checked here too.